src/main/java/MealDemo/service/MealPlanningServiceImpl.java

```diff
@@ -35,109 +43,131 @@ public List<Frequency> getFrequencies() {
 
     @Override
     public Optional<Meal> findMealById(int mealId) {
-        return mealRepository.findById(mealId);
+        var tempMeal = mealRepository.findById(mealId);
+
+        Meal meal = null;
+        if (tempMeal.isPresent()){
+            meal = tempMeal.get();
+        } else {
+            throw new ResponseStatusException(NOT_FOUND, "Invalid meal ID");
+        }
+
+        return Optional.of(meal);
     }
 
 
     @Override
     public void saveNewMeal(Meal meal) {
-        Frequency tempFreq = new Frequency();
+        var tempFreq = new Frequency();
         tempFreq.setMeal(meal);
         frequencyRepository.save(tempFreq);
         System.out.println("meal saved! ");
     }
 
+
+    @Transactional
     @Override
     public boolean deleteMeal(int mealId) {
+        boolean isDeleted = false;
 
-        Optional<Meal> tempMeal = mealRepository.findById(mealId);
+        var tempMeal = mealRepository.findById(mealId);
         if (tempMeal.isEmpty()) {
             System.out.println("Could not find meal");
         } else {
-            System.out.println("Deleting meal " + tempMeal + "with id " + mealRepository.findById(mealId));
+            System.out.println("Deleting meal " + tempMeal.get());
             frequencyRepository.deleteById(mealId);
             mealRepository.deleteById(mealId);
             System.out.println("Meal Deleted");
+            isDeleted = true;
         }
 
-        return tempMeal.isPresent();
+        return isDeleted;
     }
 
     @Override
     public void updateMeal(int mealId, Meal meal) {
         // if cant find mealId will show 404 error cant find ID
-        Meal existingMeal = mealRepository.getById(mealId);
-
-
-        System.out.println("Recipe to edit " + existingMeal.getClass().getName() + " with id " + mealId);
-
-        existingMeal.setMealName(meal.getMealName());
-        existingMeal.setCategory(meal.getCategory());
-
-        mealRepository.save(existingMeal);
-
-        System.out.println("meal has been updated. Meal details \n" + existingMeal.toString());
+        var tempMeal = mealRepository.findById(mealId);
+        if (tempMeal.isPresent()) {
+            var existingMeal = mealRepository.getById(mealId);
+            existingMeal.setMealName(meal.getMealName());
+            existingMeal.setCategory(meal.getCategory());
+            mealRepository.save(existingMeal);
+            System.out.println("meal has been updated. Meal details \n" + existingMeal.toString());
+        } else {
+            throw new ResponseStatusException(NOT_FOUND, "Invalid meal ID");
+        }
     }
 
 
     @Override
     public Map<String, List<Meal>> mealsByCategories() {
-        List<Meal> allMeals = getMeals();
+        var allMeals = getMeals();
         Map<String, List<Meal>> mealsByCategoriesList = new HashMap<>();
-        for (Meal meal : allMeals) {
+        for (var meal : allMeals) {
```

Comment:
Neat, I didn't know the `var` keyword was usable in loops too... Now I learned something too